src/firefly/js/tables/TablesCntlr.js

```diff
@@ -102,6 +102,99 @@ export const TABLE_UPDATE = `${DATA_PREFIX}.update`;
+ * @prop {string} visibility    show, hide, or hidden.  hidden columns are not viewable by users.
+ * @prop {string} sortByCols    for multi-columns sorting.  column names separated by comma(',').
+ * @prop {string} related       highlight related rows based on this column's value.
+ */
+
+/**
+ * Table meta information.  Below is only a small set of predefined meta used by table.
+ * The meta information in this object are used by many components for many reasons.  ie catalog overlay.
+ * @typedef {object} TableMeta
+ * @prop {string} Loading-Status COMPLETED or INPROGRESS
+ * @prop {string} tblFilePath   path of the source of this table on the server-side.
+ * @prop {string} isFullyLoaded 'true' when table is completely loaded on the server-side.
+ * @prop {string} source    path of the original table source before any operations were performed. ie sort, filter, etc.  this may not be fully supported.
+ */
+
+/**
+ * Table request.  Below is a list of predefined parameters available for table request.  All of the options are optional.  
+ * These parameters let you control what data and how it will be returned.
+ * @typedef {object} TableRequest
+ * @prop {number} startIdx  the starting index to fetch.  defaults to zero.
+ * @prop {number} pageSize  the number of rows per page.  defaults to 100.
+ * @prop {string} filters   list of conditions separted by comma(,). Format:  (col_name|index) operator value.
+ *                  operator is one of '> < = ! >= <= IN'.  See DataGroupQueryStatement.java doc for more details.
+ * @prop {string} sortInfo  sort information.  Format:  (ASC|DESC),col_name[,col_name]*
+ * @prop {string} inclCols  list of columns to select.  Column names separted by comma(,)
+ * @prop {string} decimate  decimation information.
+ * @prop {object} META_INFO meta information passed as key/value pair to server then returned as tableMeta.
+ * @prop {string} use       one of 'catalog_overlay', 'catalog_primary', 'data_primary'.
+ * @prop {string} tbl_id    a unique id of a table. auto-create if not given.
+ */
+
+/**
+ * Table group.  Define a group of tables used by the UI.
+ * @typedef {Object} TableGroup
+ * @prop {string}   name     unique name of this group
+ * @prop {string}   active   tbl_id of the active table in this group
+ * @prop {Object.<string, TableGroupItem>}   tables     a map of TableGroupItem(s) keyed by tbl_id
+ */
+
+/**
+ * Table group item.  Contains enough key information to identify the table data as well as the UI data associate with this item.
+ * @typedef {Object} TableGroupItem
+ * @prop {string}   tbl_group  table group name
+ * @prop {string}   tbl_id     unique id of the table data
+ * @prop {string}   tbl_ui_id  unique id of the table's UI data
+ * @prop {string}   title      title or label of the table
+ * @prop {boolean}  removable  true if this item can be removed from group.
+ * @prop {Object.<string, *>}   options   table options, ie.  selectable, expandable
+ */
+
+
```

Comment:
JSDoc should be able to generate docs with `@typedef` and `@callback` tags.  They are standard tags.